addons/survey/models/survey_question.py

```diff
@@ -264,6 +265,36 @@ def _compute_save_as_email(self):
             if question.question_type != 'char_box' or not question.validation_email:
                 question.save_as_email = False
 
+    @api.depends('survey_id.session_speed_rating_time_limit', 'survey_id.session_speed_rating', 'is_time_limited')
+    def _compute_time_limit(self):
+        """If time limit value is set manually on the question, we do not update it
+        when changing the time limit value of the survey or disabling it.
+        """
+        for question in self:
+            if not question.is_time_limited:
+                question.time_limit = False
+            if not question.survey_id.session_speed_rating:
+                if question.time_limit == question.survey_id.session_speed_rating_time_limit:
+                    question.time_limit = False
+                    question.is_time_limited = False
```

Comment:
Don't really understand ... if you remove the time limit, either we remove it on all questions, either we don't do it, but I don't really like that kind of heuristics because you have incomplete changes. Especially those are two separate settings in my opinion: limited questions, and rewards based on speed. If you want to remove time limits, you have to update it in question views, seems better to me than magic stuff.

Reply:
The reason I did it before was the following logic: We can have a time limit on separate questions without activating the rewards based on speed. If Someone already configured the question time limit, and then activated the rewards button, the question time limit that was already there in my mind should not change. Similarly, if they deactivate the rewards, time limits that were already there before activating reward option should not get deactivated. 

We could simply have every question time limit updated based on the survey "quick rewards" time limit, but that would mean that if user wants to have special time limits on specific questions, they will have to go back to these questions and update them every time they change survey quick rewards time limit

Reply:
Ok, actually the feature is better with more use case covered, indeed.